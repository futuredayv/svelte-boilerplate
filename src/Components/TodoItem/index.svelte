<script>
	export let todo = {};

	$: completed = todo.status;
</script>

<style src="./style.scss">

</style>

<div class="todo-item" class:completed>
	<div class="todo-item__status">
		<input type="checkbox" bind:checked={completed}>
	</div>

	<div class="todo-item__title">{todo.title}</div>
</div>
