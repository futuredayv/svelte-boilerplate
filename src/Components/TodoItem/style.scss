.todoItem {
    
    &.completed {
        color: rgba(0, 0, 0, 0.4);
        text-decoration: overline;
    }
}