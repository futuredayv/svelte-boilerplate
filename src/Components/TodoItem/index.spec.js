import TodoItem from './index.svelte';
import { render } from '@testing-library/svelte';

describe('Components', () => {
	describe('TodoItem', () => {
		xit('renders without failing', () => {
	
			const todo = {
				name: 'Run Tests',
				status: false,
			};
			
			const { container } = render(<TodoItem todo={todo}></TodoItem>)
	
			expect(container.querySelector('.todo-item').length).toBeTruthy();
			expect(container).toMatchSnapshot()
		});
	})
});