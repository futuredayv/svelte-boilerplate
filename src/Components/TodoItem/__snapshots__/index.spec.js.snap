// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample renders without failing 1`] = `
<body>
  <div>
    <div
      class="todoItem"
    >
      Hello World!
    </div>
  </div>
</body>
`;
